### PR TITLE
echo websocket stream error

### PR DESCRIPTION
**Pull request by SyMind**

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -279,8 +279,10 @@ MqttClient.prototype._setupStream = function () {
 
   this.stream.pipe(writable)
 
-  // Suppress connection errors
-  this.stream.on('error', nop)
+  // Echo connection errors
+  this.stream.on('error', function (error) {
+    that.emit('error', error)
+  })
```
Comment (mcollina) [inline lib/client.js]: Ignoring those errors is done on purpose. I’d be ok in emitting them as “streamError” instead.

File: lib/connect/wx.js [hunk 2]
```diff
@@ -104,6 +104,27 @@ function buildStream (client, opts) {
 
   proxy = buildProxy()
   stream = duplexify.obj()
+  stream._destroy = function (err, cb) {
+    socketTask.close({
+      success: function () {
+        cb && cb(err)
+      }
```
Comment (mcollina) [inline lib/connect/wx.js]: You might want to call the callback in case or error as well.

File: lib/connect/wx.js [hunk 3]
```diff
@@ -104,6 +104,27 @@ function buildStream (client, opts) {
...
+
+    var self = this
+    process.nextTick(function () {
+      socketTask.close({
+        fail: function () {
+          self._destroy(new Error())
+        }
+      })
+    })
+  }.bind(stream)
```
Comment (mcollina) [inline lib/connect/wx.js]: I don’t think this method should be overridden.

File: lib/client.js (hunk 1 above)
Comment (SyMind) [reply lib/client.js]: When will it be updated code?

File: lib/connect/wx.js (hunk 2 above)
Comment (SyMind) [reply lib/connect/wx.js]: sorry, it cause by donot merge code from master.

**Pull request closed without merging**